pdns/lua-auth4.cc

```diff
@@ -222,8 +224,45 @@ AuthLua4::AuthLua4(const std::string& fname) {
   d_lw->executeCode(ifs);
 
   d_update_policy = d_lw->readVariable<boost::optional<luacall_update_policy_t>>("updatepolicy").get_value_or(0);
+  d_axfr_filter = d_lw->readVariable<boost::optional<luacall_axfr_filter_t>>("axfrfilter").get_value_or(0);
+
 }
 
+bool AuthLua4::axfrfilter(const ComboAddress& remote, const DNSName& zone, const DNSResourceRecord& in, vector<DNSResourceRecord>& out) {
+  luacall_axfr_filter_t::result_type ret;
+
+  if (d_axfr_filter == NULL) return false;
+
+  ret = d_axfr_filter(remote, zone, in);
+
+  if (std::get<0>(ret) < 0)
+    return false;
+  else if (std::get<0>(ret) > 0)
```

Comment:
maybe do `== 1` here so we can later give 2 meaning as well. Also it might be good to make these constants that the code can use, just like `pdns.PASS`